lib/logstash/outputs/s3.rb

```diff
@@ -357,16 +414,31 @@ def configure_periodic_rotation
       LogStash::Util::set_thread_name("<S3 periodic uploader")
 
       Stud.interval(periodic_interval, :sleep_then_run => true) do
-        @logger.debug("S3: time_file triggered, bucketing the file", :filename => @tempfile.path)
-
-        move_file_to_bucket_async(@tempfile.path)
-        next_page
-        create_temporary_file
+        @tempfile.keys.each do |key|
+          @logger.debug("S3: time_file triggered, bucketing the file", :filename => @tempfile[key].path)
+          move_file_to_bucket_async(@tempfile[key].path)
+          next_page(key)
+          create_temporary_file(key)
+        end
       end
     end
   end
 
   private
+  def cleanprefix(prefix)
```

Comment:
Can you rename `cleanprefix` for `clean_prefix`, we usually try to split words with underscore.